client/state/preferences/actions.js

```diff
@@ -53,17 +70,29 @@ export const setPreference = ( key, value ) => ( {
  */
 export const savePreference = ( key, value ) => dispatch => {
 	dispatch( setPreference( key, value ) );
+	dispatch( {
+		type: PREFERENCES_SAVE,
+		key,
+		value
+	} );
+
 	const settings = {
 		[ USER_SETTING_KEY ]: {
 			[ key ]: value
 		}
 	};
 
-	return wpcom.me().settings().update( JSON.stringify( settings ) )
-		.then( ( data ) => {
-			dispatch( {
-				type: PREFERENCES_RECEIVE,
-				data
-			} );
+	return wpcom.me().settings().update( settings ).then( ( data ) => {
```

Comment:
Do we need the `JSON.stringify` here? When I dismiss the media modal help forever, this is set to `'true'` rather than `true`:
<img width="464" alt="screen shot 2016-08-19 at 4 12 50 pm" src="https://cloud.githubusercontent.com/assets/1270189/17826689/4de91574-6628-11e6-89aa-d42f91354889.png">